lib/rack/cache/meta_store.rb

```diff
@@ -64,8 +64,7 @@ def store(request, response, entity_store)
         end
         response.headers['X-Content-Digest'] = digest
         response.headers['Content-Length'] = size.to_s unless response.headers['Transfer-Encoding']
-        # If the entitystore backend is a Noop, do not try to read the body from the backend, it always returns an empty array
-        response.body = entity_store.open(digest) || response.body unless entity_store.is_a? Rack::Cache::EntityStore::Noop
+        response.body = entity_store.open(digest) || response.body
```

Comment:
I think you are right, I just read this as `ah we are reading back the cached value` ... pointless indeed :(